csharp/src/Google.Protobuf/UnknownFieldSet.cs

```diff
@@ -75,7 +75,20 @@ public void WriteTo(CodedOutputStream output)
             {
                 entry.Value.WriteTo(entry.Key, output);
             }
-        }
+        }
+
+#if GOOGLE_PROTOBUF_SUPPORT_SYSTEM_MEMORY

```

Comment:
in UnknownFieldSetTest.cs there's a bunch of tests that could be also use the CodedInputReader/CodedOuputWriter codepath:
TestMergeCodedInput
TestReadInvalidWireTypeThrowsInvalidProtocolBufferException
TestDiscardUnknownFields

and perhaps a few more.